### PR TITLE
Creation of unix permission atrribute module and its test

### PR DESCRIPTION
**Pull request by Vojtech-Sassmann**
This class checks if the attribute value is correct unix permission mask. If not, exception is thrown.

### CODE REVIEW

**Comment by licehammer** [comment]
Code is OK.

Before the merge to upstream, please discard the unnecessary merge commit and either squash the other two commits into a single one or rewrite the commit messages to fit the content of the commits.

**Comment by Vojtech-Sassmann** [comment]
I have just fixed the problem with commits.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_homeDirUmask.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.FacilityAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.FacilityAttributesModuleImplApi;
+
+/**
+ * Checks if the value is valid unix permission mask.
+ *
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+public class urn_perun_entityless_attribute_def_def_homeDirUmask extends FacilityAttributesModuleAbstract implements FacilityAttributesModuleImplApi {
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_homeDirUmask.java]: If this supposed to be facility attribute, class name must match it, right now it references "entityless" attribute, which is wrong. Please note, that during class instantiation, they are referenced by exact attribute URN, converted to class name string. So for ``urn:perun:facility:attribute-def:def:homeDirUmask`` you must have ``urn_perun_facility_attribute_def_def_homeDirUmask`` module.

Please fix test class name too.

**Comment by zlamalp** [comment]
Just generic note. I believe, that ``umask != permissions``, we probably should update attribute displayName and description in a code and in production instances, @licehammer ?

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_homeDirUmask.java (hunk 1 above)
Comment (Vojtech-Sassmann) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_homeDirUmask.java]: Class names should be ok now.

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ */
+public class urn_perun_facility_attribute_def_def_homeDirUmaskTest {
+
+	private static PerunSessionImpl session;
+	private static Facility facility;
+	private static urn_perun_entityless_attribute_def_def_homeDirUmask classInstance;
+
+	@Before
+	public void setUp() {
+		classInstance = new urn_perun_entityless_attribute_def_def_homeDirUmask();
```
Comment (zlamalp) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java]: Please fix class name here too.

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Test of unix permission mask attribute.
+ *
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+public class urn_perun_facility_attribute_def_def_homeDirUmaskTest {
+
+	private static PerunSessionImpl session;
+	private static Facility facility;
+	private static urn_perun_entityless_attribute_def_def_homeDirUmask classInstance;
```
Comment (zlamalp) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java]: Please rename variable too.

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java (hunk 2 above)
Comment (Vojtech-Sassmann) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeDirUmaskTest.java]: Ok, now it should be fine.

**Pull request merged**